### PR TITLE
MediaDevices | fix apiref & add securecontext_header & some other fix

### PR DESCRIPTION
**Pull request by skyclouds2001**

### CODE REVIEW

File: files/en-us/web/api/mediadevices/enumeratedevices/index.md [hunk 1]
```diff
@@ -26,19 +26,22 @@ None.
...
 ### Return value
 
-A {{ jsxref("Promise") }} that is fulfilled with an array of {{domxref("MediaDeviceInfo")}} objects.
+A {{jsxref("Promise")}} that is fulfilled with an array of {{domxref("MediaDeviceInfo")}} objects.
 Each object in the array describes one of the available media input and output devices.
 The order is significant — the default capture devices will be listed first.
 
 Other than default devices, only devices for which permission has been granted are "available".
 
+If the media device is an input device, a {{domxref("InputDeviceInfo")}} object which extends {{domxref("MediaDeviceInfo")}} will be returned instead.
```
Comment (hamishwillee) [inline files/en-us/web/api/mediadevices/enumeratedevices/index.md:35]: "an". Don't think we need to talk about the fact it extends MediaDeviceInfo here.
```suggestion
If the media device is an input device, an {{domxref("InputDeviceInfo")}} object will be returned instead.
```

File: files/en-us/web/api/mediadevices/enumeratedevices/index.md [hunk 2]
```diff
@@ -26,19 +26,22 @@ None.
...
+
 If enumeration fails, the promise is rejected.
 
 ## Security requirements
 
 Access to the API is subject to the following constraints:
 
 - The method must be called in a [secure context](/en-US/docs/Web/Security/Secure_Contexts).
+- The document must be fully active and its visibility must be "visible".
```
Comment (hamishwillee) [inline files/en-us/web/api/mediadevices/enumeratedevices/index.md:44]: That's a good catch!

File: files/en-us/web/api/mediadevices/getsupportedconstraints/index.md [hunk 3]
```diff
@@ -6,10 +6,9 @@ page-type: web-api-instance-method
 browser-compat: api.MediaDevices.getSupportedConstraints
 ---
 
-{{APIRef("Media Capture and Streams")}}
+{{APIRef("Media Capture and Streams")}}{{securecontext_header}}
```
Comment (hamishwillee) [inline files/en-us/web/api/mediadevices/getsupportedconstraints/index.md:9]: Why is this secure context? https://w3c.github.io/mediacapture-main/#webidl-823932458 doesn't show it as such?

File: files/en-us/web/api/mediadevices/index.md [hunk 4]
```diff
@@ -5,9 +5,9 @@ page-type: web-api-interface
 browser-compat: api.MediaDevices
 ---
 
-{{APIRef("Media Capture and Streams")}}
+{{APIRef("Media Capture and Streams")}}{{securecontext_header}}
```
Comment (hamishwillee) [inline files/en-us/web/api/mediadevices/index.md:8]: My understanding is that this should only be secure context if ALL the methods/properties require secure context, which I am not sure is the case for the getSupportedConstraints and friend here https://w3c.github.io/mediacapture-main/#webidl-823932458.

Upshot, possibly this goes. If it does, we need to add secure context inline on the ones that are secure context in this page.

File: files/en-us/web/api/mediadevices/getsupportedconstraints/index.md (hunk 3 above)
Comment (skyclouds2001) [reply files/en-us/web/api/mediadevices/getsupportedconstraints/index.md:9]: yeah, but the `MediaDevices` interface requires secure context, and the `Navigator.mediaDevices` also requires secure context, then `MediaDevices.getSupportedConstraints()` surely requires secure context

Comment (skyclouds2001) [reply files/en-us/web/api/mediadevices/getsupportedconstraints/index.md:9]: for those partical interface definition, some time the flags like secure context may only list in the main interface definition, not in the partical interface definition

File: files/en-us/web/api/mediadevices/index.md (hunk 4 above)
Comment (skyclouds2001) [reply files/en-us/web/api/mediadevices/index.md:8]: the https://w3c.github.io/mediacapture-main/#dom-mediadevices says it requires secure context

Comment (wbamberg) [reply files/en-us/web/api/mediadevices/index.md:8]: Yeah, I think @skyclouds2001 is right, if the main interface is marked secure context then partials and are implicitly secure context as well, as are any mixins included in it.

**Comment by wbamberg** [comment]
Since @hamishwillee is on this, I'm removing my review request, but please feel free to @ me if you want to talk about anything in it 👍 .

**Review by hamishwillee — APPROVED** [review]
Thank you

**Pull request merged**